provisioning/setup.py

```diff
@@ -83,14 +101,214 @@ def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            if check_cluster_status() != 0:
+                logging.error('Cluster status reports failure')
+                exit(-1)
+        except Exception as error:
+            logging.error('Error while checking cluster status (%s)',
+                          error)
+            exit(-1)
+
+class SupportBundle(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            # Default target directory is /tmp/hare
+            if os.system('hctl reportbug') != 0:
+                logging.error('Failed to generate support bundle')
+                exit(-1)
+        except Exception as error:
+            logging.error('Error while generating support bundle (%s)',
+                          error)
+            exit(-1)
+
+# Approach: Clear hare logs generated until now
+class Cleanup(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            if os.system('rm -f /var/motr/hax/*') != 0:
+                logging.error('Failed to clear Hare logs')
+                exit(-1)
+        except Exception as error:
+            logging.error('Failed to clear Hare logs (%s)',
+                          error)
+            exit(-1)
+
+def checkRpm(rpm_name):
+    return os.system('rpm -qa | grep -q ' + rpm_name)
+
+
+def bootstrap_cluster():
+    return os.system('hctl bootstrap --mkfs /var/lib/hare/cluster.yaml')
+
+
+def shutdown_cluster():
+    return os.system('hctl shutdown')
+
+
+def check_cluster_status():
+    cluster_desc = None
+    with open("/var/lib/hare/cluster.yaml", 'r') as stream:
+        cluster_desc = yaml.safe_load(stream)
+    hostname = socket.gethostname()
+
+    # Finding info about current node from CDF based on hostname
+    for node in cluster_desc['nodes']:
+        if node['hostname'] == hostname:
+            m0ds = node.get('m0_servers', [])
+
+            # We will calculate number of IO services, m0clients, presence of confd based on CDF
+            # and then we will read 'hctl status --json' and compare number of actual services running
+            # to determine whether cluster is configured and running correctly
+            confd_p_cdf = False
+            confd_p = False
+            ios_count = 0
+            ios_count_cdf = 0
+            for m0d in m0ds:
+                if 'runs_confd' in m0d.keys():
+                    if m0d['runs_confd']:
+                        confd_p_cdf = True
+                if m0d['io_disks']['data']:
+                    ios_count_cdf  += 1
+            m0client_count_cdf = node['m0_clients']['s3']
+            m0client_count_cdf += node['m0_clients']['other']
+            m0client_count = 0
+
+            cmd = 'hctl status --json'
+            cluster_info = json.loads(execute_command_and_get_output(cmd))
+            nodes_data = cluster_info['nodes']
+
+            for nodeinfo in nodes_data:
+                if nodeinfo[0] == hostname:
```

Comment:
@SwapnilGaonkar7 I'm afraid you're relying on a broken JSON format. 

It is fixed by https://github.com/Seagate/cortx-hare/pull/1397
@mssawant would you mind to review it finally? 😄

Reply:
Merged!

Reply:
@SwapnilGaonkar7 please merge the freshest `main` branch into your PR and accommodate the changes in `hctl status --json` output (an example can be also seen in #1397 description). Thanks!

Reply:
Done